src/api/main/NativeInterfaceFactory.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+import type {NativeInterfaceMain} from "../../native/main/NativeInterfaceMain"
+import type {NativePushServiceApp} from "../../native/main/NativePushServiceApp"
+import type {NativeSystemApp} from "../../native/main/NativeSystemApp"
+import type {NativeFileApp} from "../../native/common/FileApp"
+import {downcast} from "@tutao/tutanota-utils"
+import {isBrowser} from "../common/Env"
+import {ProgrammingError} from "../common/error/ProgrammingError"
+
+/**
+ * A factory function to create interfaces for native communication
+ * When running in browser, these interfaces will be stubs that throw if you try to use them
+ */
+export async function createNativeInterfaces(): Promise<{
+	native: NativeInterfaceMain,
+	fileApp: NativeFileApp,
+	pushService: NativePushServiceApp,
+	systemApp: NativeSystemApp
+}> {
+
+	let native
+	let fileApp
+	let pushService
+	let systemApp
+
+	if (!isBrowser()) {
+		const {NativeInterfaceMain} = await import("../../native/main/NativeInterfaceMain")
```

Comment:
importing all of it during locator init probably slows down some things but I hope they are small enough so that it makes little difference